yt_dlp/downloader/hls.py

```diff
@@ -28,7 +28,14 @@ class HlsFD(FragmentFD):
     FD_NAME = 'hlsnative'
 
     @staticmethod
-    def can_download(manifest, info_dict, allow_unplayable_formats=False):
+    def _has_drm(manifest):  # TODO: https://github.com/yt-dlp/yt-dlp/pull/5039
+        return bool(re.search('|'.join((
+            r'#EXT-X-(?:SESSION-)?KEY:.*?URI="skd://',  # Apple FairPlay
```

Comment:
maybe we could check like this? what do you think
```suggestion
            r'KEYFORMAT="com.apple.streamingkeydelivery"',  # Apple FairPlay
            r'KEYFORMAT="com.microsoft.playready"',  # Microsoft PlayReady
```

Reply:
We only want to match lines starting with `#EXT-X-(?:SESSION-)?KEY:`

Reply:
```suggestion
            r'#EXT-X-(?:SESSION-)?KEY:.*?URI="skd://',  # Apple FairPlay
            r'#EXT-X-(?:SESSION-)?KEY:.*?KEYFORMAT="com.apple.streamingkeydelivery"',  # Apple FairPlay
            r'#EXT-X-(?:SESSION-)?KEY:.*?KEYFORMAT="com.microsoft.playready"',  # Microsoft PlayReady
```

Reply:
I dunno if the `URI="skd://` pattern would still be necessary since `skd` == `streamingkeydelivery` but no harm in keeping it ig

Reply:
Yeah, I don't want to remove it just in case.